File: rust/src/shared/traits/executable.rs
pub trait Executable {
    fn execute(&self, chapter: u32) -> anyhow::Result<()> {
        match chapter {
            1 => self.question_1(),
            2 => self.question_2(),
            3 => self.question_3(),
            4 => self.question_4(),
            _ => Ok(()),
        }
    }

    fn question_1(&self) -> anyhow::Result<()>;
    fn question_2(&self) -> anyhow::Result<()>;
    fn question_3(&self) -> anyhow::Result<()>;
    fn question_4(&self) -> anyhow::Result<()>;
}

File: TypeScript/shared/Executable.ts
export default abstract class Executable {
    abstract question1(): void;
    abstract question2(): void
    abstract question3(): void;
    abstract question4(): void;

    execute(num: number): void {
        switch (num) {
            case 1:
                this.question1();
                break;
            case 2:
                this.question2();
                break;
            case 3:
                this.question3();
                break;
            case 4:
                this.question4();
                break;
            default:
                break;
        }
    }
}